src/zcash/Proof.hpp

```diff
@@ -244,6 +244,12 @@ class ProofVerifier {
     ProofVerifier(bool perform_verification) : perform_verification(perform_verification) { }
 
 public:
+    // ProofVerifier should never be copied
```

Comment:
It's not obvious to me why this is the case. It seems like a "value type" rather than a "singleton-ish / no-copy" type, simply based on the structure.

Should it be no-copy because:
* it or a subclass or a future revision may depend on shared state that should not be copied and also shouldn't be aliased (ie accessed via multiple pointers/references)?
* any call site must always be able to use it without copying, so this is just a stylistic enforcement on call sites?
* some other reason?